pkg/types/types.go

```diff
@@ -12,8 +12,10 @@ import (
 
 // ResourceIdentifier provides the identifiers needed to interact with any arbitrary object.
 type ResourceIdentifier struct {
-	GroupVersionKind schema.GroupVersionKind
-	NamespacedName   types.NamespacedName
+	Ordinal              int
```

Comment:
since it is a list in work api, and we transfer to a map then apply to spoke. I think we have to keep the index of the manifest in that list during this process. Then when we get the apply result, we would be able to update the manifest status in work.Status.